scripts/earthengine/process_events.py

```diff
@@ -531,24 +563,67 @@ def deactivate_old_events(self, event_ids: list, date: str) -> list:
                 active_ids.append(event_id)
             else:
                 # Event ended too long ago. Remove from active s2 cells.
-                _DEBUG and logging.debug(
+                logging.level_debug() and logging.debug(
                     f'Deactivating event: {event_id} with end date older than {min_date}'
                 )
-                self.remove_event_from_place_index(event_id)
+                event = self.get_event_by_id(event_id)
+                if event:
+                    places = event.get_places()
+                    self.remove_event_from_place_index(event_id)
+                    self._counters.add_counter('events_deactivated', 1,
+                                               event_id)
+                    self._counters.add_counter('events_places_deactivated',
+                                               len(places), event_id)
         return active_ids
 
+    def add_event_to_place_index(self, event: GeoEvent):
+        '''Add places for an event into the lace index.'''
+        if not event:
+            return
+        event_id = event.event_id()
+        places = event.get_places().keys()
+        logging.level_debug() and logging.debug(
+            f'Adding event {event_id} to place index for: {places}')
+        for place_id in places:
+            self._active_event_by_place[place_id] = event
+
     def remove_event_from_place_index(self, event_id: str):
         '''Remove the s2 cells for this event from the _active_event_by_place index.'''
         event = self.get_event_by_id(event_id)
+        if not event:
+            return
         places = event.get_places().keys()
-        _DEBUG and logging.debug(
+        logging.level_debug() and logging.debug(
             f'Removing event {event_id} active places: {places}')
         for place_id in places:
-            if place_id in self._active_event_by_place:
+            place_event = self._active_event_by_place.get(place_id, None)
+            if place_event and place_event.event_id() == event_id:
                 self._active_event_by_place.pop(place_id)
-        self._counters.add_counter('events_deactivated', 1, event_id)
-        self._counters.add_counter('events_places_deactivated', len(places),
-                                   event_id)
+
+    def merge_events(self, event_ids: list) -> GeoEvent:
+        '''returns root_event, which is the earliest event in the events_list and 
```

Comment:
What is "root_event"?

Reply:
Added some more comments. Pls let me know of any alternate phrase that is easier to follow :)